subsys/bluetooth/shell/bt.c

```diff
@@ -455,18 +461,30 @@ static int cmd_hci_cmd(const struct shell *shell, size_t argc, char *argv[])
 	uint16_t ocf;
 	struct net_buf *buf = NULL, *rsp;
 	int err;
+	static uint8_t hex_data[HCI_CMD_MAX_PARAM];
+	int hex_data_len;
 
+	hex_data_len = 0;
 	ogf = strtoul(argv[1], NULL, 16);
 	ocf = strtoul(argv[2], NULL, 16);
 
 	if (argc > 3) {
-		size_t i;
+		size_t len;
 
-		buf = bt_hci_cmd_create(BT_OP(ogf, ocf), argc - 3);
+		if (strlen(argv[3]) > 2 * HCI_CMD_MAX_PARAM) {
+			shell_error(shell, "Data field too large\n");
+			return -ENOEXEC;
+		}
 
-		for (i = 3; i < argc; i++) {
-			net_buf_add_u8(buf, strtoul(argv[i], NULL, 16));
+		len = hex2bin(argv[3], strlen(argv[3]), &hex_data[hex_data_len],
```

Comment:
hex_data_len is just zero, you could get rid of this variable IMO.

Reply:
Yes, considered that ,but left it, in case we want to split up the parameters send